test/test_serialization.py

```diff
@@ -568,6 +568,28 @@ def test_load_error_msg(self):
         with self.assertRaisesRegex(AttributeError, expected_err_msg):
             torch.load(resource)
 
+    def test_save_different_dtype_unallocated(self):
+        devices = ['cpu']
+        if torch.cuda.is_available():
+            devices.append('cuda')
+
+        for device in devices:
+            a = torch.tensor([], dtype=torch.complex128, device=device)
```

Comment:
looks good although I think it might be better to add a test that tests this for all dtypes since that was the reported bug!

Reply:
Good point. I've updated to test all dtypes and also check the result of `torch.load`